primitives/runtime/src/offchain/storage_lock.rs

```diff
@@ -0,0 +1,397 @@
+// Copyright 2019-2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! # Offchain Storage Lock
+//!
+//! A storage-based lock with a defined expiry time.
+//!
+//! The lock is using Local Storage and allows synchronizing
+//! access to critical section of your code for concurrently running Off-chain Workers.
+//! Usage of `PERSISTENT` variant of the storage persists the lock value even in case of re-orgs.
+//!
+//! A use case for the lock is to make sure that a particular section of the code is only run
+//! by one Off-chain Worker at the time. This may include performing a side-effect (i.e. an HTTP call)
+//! or alteration of single or multiple Local Storage entries.
+//!
+//! One use case would be collective updates of multiple data items
+//! or append / remove of i.e. sets, vectors which are stored in
+//! the offchain storage DB.
+//!
+//! ## Example:
+//!
+//! ```rust
+//! // in your off-chain worker code
+//!
+//! fn append_to_in_storage_vec<'k, T>(key: &'k [u8], T) where T: Encode {
+//!    let mut lock = StorageLock::new(b"x::lock");
```

Comment:
Without reading anything else in the PR other than this module doc I find it a bit confusing to know what `x::lock` here refers to. So this must refer to some key in the offchain storage? In this case, why is it not related to `key`?

Or does it mean that any code trying to access the same `x::lock` would be blocked here?

Reply:
`x::lock` refers to the lock identifier, but I'll adjust the docs in order to be more explicit.